test/common/config/udpa_resource_test.cc

```diff
@@ -4,6 +4,17 @@
 
 #include "gtest/gtest.h"
 
+using ::testing::ElementsAre;
+using ::testing::Pair;
+using ::testing::UnorderedElementsAre;
+
+#define EXPECT_CONTEXT_PARAMS(context_params, ...)                                                 \
+  {                                                                                                \
+    std::map<std::string, std::string> param_map((context_params).begin(),                         \
```

Comment:
Why is it necessary to convert to `std::map<>` here?  Doesn't the protobuf map field accessor provide the same interface as `std::map<>`, such that it works directly with `UnorderedElementsAre()`?

I would think that you could just say `EXPECT_THAT(resource_name.context().params(), UnorderedElementsAre(...))`.

Reply:
That doesn't work, since what you get is a comparison on the proto `MapPair` object, which does not pretty-print on failure, i.e. you see stuff like:
```
Expected: has 2 elements and there exists some permutation of elements such that:
 - element #0 has a first field that is equal to "%#[]&=", and has a second field that is equal to "bar", and
 - element #1 has a first field that is equal to "foo", and has a second field that is equal to "%#[]&="
  Actual: { 48-byte object <06-66 6F-6F 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 0C-25 23-5B 5D-26 3D-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00>, 48-byte object <10-25 23-5B 5D-26 3D-61 62-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 14-63 64-65 25-23 5B-5D 26-3D 66-00 00-00 00-00 00-00 00-00 00-00 00-00>, 48-byte object <0C-25 23-5B 5D-26 3D-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 06-62 61-72 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00> }, which has 3 elements
```

Reply:
Ah, okay.  That's too bad.

Might be worth adding a comment about why this conversion is necessary.